Code/src/main/java/nl/utwente/viskell/ui/components/Block.java

```diff
@@ -108,6 +93,7 @@ protected void createCircleMenu() {
     private void handleMouseEvent(MouseEvent t) {
         parentPane.setSelectedBlock(this);
         if (t.getButton() == MouseButton.SECONDARY) {
+            CircleMenu circleMenu = new CircleMenu(this);
```

Comment:
This introduces a regression where right-clicking multiple times on a block creates multiple circle menu's.